### PR TITLE
Add clarifying parenthesis around non-trivial conditions in ternary expressions.

### LINKED ISSUE #85868: llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp:3804: lacking () for clarity
Static analyser cppcheck says:

llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp:3804:47: style: Clarify calculation precedence for '&' and '?'. [clarifyCalculation]

Source code is

    for (unsigned i = TSFlags & X86II::EVEX_K ? 2 : 1;

I am guessing that 

    for (unsigned i = (TSFlags & X86II::EVEX_K) ? 2 : 1;

was intended. Suggest add () for clarity of intention. 
Not all of us know all the C precedence rules well.

### PR DESCRIPTION
**Pull request by luolent**
Fixes [#85868](https://github.com/llvm/llvm-project/issues/85868)

Parenthesis are added as requested on ternary operators with non trivial conditions.
I used this [precedence table](https://en.cppreference.com/w/cpp/language/operator_precedence) for reference, to make sure we get the expected behavior on each change.

### CODE REVIEW

**Review by RKSimon — COMMENTED** [review]
Please update the PR subject as its a lot more than just X86AsmParser.cpp

**Review by RKSimon — COMMENTED** [review]
Please address the clang-format warnings the CI has reported

**Comment by luolent** [comment]
> Please update the PR subject as its a lot more than just X86AsmParser.cpp

Hi @RKSimon 

All the issues mentioned are fixed. The tile of the PR is misleading. 
The title is the same as the issue (#85868) it corresponds to.
Got a look to other PR's and I thought that this is the usual naming convention.
Please refer to the [commit](https://github.com/llvm/llvm-project/pull/90391/commits/54c6c6b7d71f5ff293412f5f91e9f880480284f8) and you will see all the modified files.

**Comment by joker-eph** [comment]
PR should be named according to what they are actually achieving, I'm not sure why the GitHub issue title is relevant?

**Comment by bulbazord** [comment]
> > Please update the PR subject as its a lot more than just X86AsmParser.cpp
> 
> Hi @RKSimon
> 
> All the issues mentioned are fixed. The title of the PR is misleading. The title is the same as the issue (#85868) it corresponds to. Got a look to other PR's and I thought that this is the usual naming convention. Please refer to the [commit](https://github.com/llvm/llvm-project/pull/90391/commits/54c6c6b7d71f5ff293412f5f91e9f880480284f8) and you will see all the modified files.

Commit titles should accurately reflect a change. Your change modifies more than just X86AsmParser.cpp, so reading the commit title alone might lead one to believe that's all that changed. But it's not, you changed many things across the code base (even if the changes are all of the same kind). A more accurate title might be something like: `Add clarifying parenthesis around non-trivial conditions in ternary expressions`.

**Review by ldionne — APPROVED** [review]
The libc++abi changes look fine to me.

**Comment by lntue** [comment]
> #85868

You can update the title of the PR to reflect the changes, and add `Fixes https://github.com/llvm/llvm-project/issues/85868` to relating the PR to the issue.

OTOH, the libc changes look good to me.

**Comment by luolent** [comment]
Hi @bulbazord , @lntue ,

thanks for your helpful comments, first time contributing here :)
I have updated the PR.

**Comment by jayfoad** [comment]
AMDGPU changes are fine.

**Review by RKSimon — APPROVED** [review]
LGTM

File: llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp [hunk 1]
```diff
@@ -3802,7 +3802,7 @@ bool X86AsmParser::validateInstruction(MCInst &Inst, const OperandVector &Ops) {
     //    VFMULCPHZrr   Dest, Src1, Src2
     //    VFMULCPHZrrk  Dest, Dest, Mask, Src1, Src2
     //    VFMULCPHZrrkz Dest, Mask, Src1, Src2
-    for (unsigned i = TSFlags & X86II::EVEX_K ? 2 : 1;
```
Comment (MaskRay) [inline llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp:3805]: I know it's subjective, but the new code with two levels of parens decreases readability.

File: lld/ELF/LinkerScript.cpp [hunk 2]
```diff
@@ -801,7 +801,7 @@ static OutputDesc *addInputSec(StringMap<TinyPtrVector<OutputSection *>> &map,
       auto *firstIsec = cast<InputSectionBase>(
           cast<InputSectionDescription>(sec->commands[0])->sectionBases[0]);
       OutputSection *firstIsecOut =
-          firstIsec->flags & SHF_LINK_ORDER
```
Comment (MaskRay) [inline lld/ELF/LinkerScript.cpp:804]: Why is random change made to lld/ELF?

`[-+*%] \w+ \?'` is extensively used and I don't see we add `)`

File: llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp (hunk 1 above)
Comment (jrtc27) [reply llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp:3805]: The second level seems like a bug, it's not consistent with how `type identifier = expr ? expr : expr;` is handled in a block.

File: lld/ELF/LinkerScript.cpp (hunk 2 above)
Comment (luolent) [reply lld/ELF/LinkerScript.cpp:804]: Hi @MaskRay ,

the PR modifies all the files with non-trivial ternary operators given by Cppcheck ([issue](https://github.com/llvm/llvm-project/issues/85868)).
You can find the Cppcheck report inside the comments thread of the issue.

Comment (MaskRay) [reply lld/ELF/LinkerScript.cpp:804]: A CppCheck issue may or may not be useful for the project. Projects have different code styles. Applying a random static analyzer report may not be useful.

**Pull request merged**